### ISSUE TITLE
lodash 3.8.0

### ISSUE THREAD

**Issue by sixinli** [cpettitt/dagre#178]
Hi,

I am wondering if dagre v0.7.2 will be compatible with lodash 3.8.0? (it's currently on 2.4.1)

Thanks,
Sixin

**Comment by cpettitt** [comment]
It does not appear to be. There are three failing tests when bumping lodash to 3.8.0.

**Comment by sixinli** [comment]
oh :( 

is there a plan to upgrade lodash anytime soon?

**Comment by sheymann** [comment]
Version is 3.9.3 now.

**Comment by sheymann** [comment]
The lib support 3.10.0, but there is a bug in the test system that fail to deeply compare arrays. Please read as follows, the expected and actual parts are the same:

  286 passing (525ms)
  4 failing

  1) version should match the version from package.json:
     AssertionError: expected {} to equal '0.7.3-pre'
      at Context.<anonymous> (test/version-test.js:6:39)

  2) acyclic greedy run does not change an already acyclic graph:

      AssertionError: expected [ Array(4) ] to deeply equal [ Array(4) ]
      + expected - actual

           "v": "a"
           "w": "b"
         }
         {
      +    "v": "a"
      +    "w": "c"
      +  }
      +  {
           "v": "b"
           "w": "d"
         }
         {
      -    "v": "a"
      -    "w": "c"
      -  }
      -  {
           "v": "c"
           "w": "d"
         }
       ]
      
      at Context.<anonymous> (test/acyclic-test.js:32:52)

  3) acyclic dfs run does not change an already acyclic graph:

      AssertionError: expected [ Array(4) ] to deeply equal [ Array(4) ]
      + expected - actual

           "v": "a"
           "w": "b"
         }
         {
      +    "v": "a"
      +    "w": "c"
      +  }
      +  {
           "v": "b"
           "w": "d"
         }
         {
      -    "v": "a"
      -    "w": "c"
      -  }
      -  {
           "v": "c"
           "w": "d"
         }
       ]
      
      at Context.<anonymous> (test/acyclic-test.js:32:52)

  4) acyclic unknown-should-still-work run does not change an already acyclic graph:

      AssertionError: expected [ Array(4) ] to deeply equal [ Array(4) ]
      + expected - actual

           "v": "a"
           "w": "b"
         }
         {
      +    "v": "a"
      +    "w": "c"
      +  }
      +  {
           "v": "b"
           "w": "d"
         }
         {
      -    "v": "a"
      -    "w": "c"
      -  }
      -  {
           "v": "c"
           "w": "d"
         }
       ]
      
      at Context.<anonymous> (test/acyclic-test.js:32:52)

**Comment by sixinli** [comment]
cool, maybe we can try to fix the tests if that's the only thing blocking the update?

**Comment by cpettitt** [comment]
@sheymann thanks for digging into it.

I looked at the test, and it looks like it is written correctly:

```js
          expect(_.sortBy(results, ["v", "w"])).to.eql([
            { v: "a", w: "b" },
            { v: "a", w: "c" },
            { v: "b", w: "d" },
            { v: "c", w: "d" }
          ]);
```

The test expects that the results, sorted by "v" and then by "w", would equal the sorted array specified. It seems that the previous behavior for `_.sortBy` has changed in a way that is not entirely clear, as it is no longer returning the results sorted.

A simple test confirms:

```js
_.sortBy([{ v: "a", w: "b" }, { v: "c", w: "d" }, { v: "a", w: "e"}], ["v", "w"])
// =>
// [ { v: 'a', w: 'b' },
//  { v: 'c', w: 'd' },
//  { v: 'a', w: 'e' } ]
```

It's possible I was misusing `_.sortBy`, but it's not obvious on quick inspection how. Once we figure this out we need to do a pass and make sure `_.sortBy` is not used incorrectly elsewhere, in a way that is not detected by a test.

**Comment by cpettitt** [comment]
Looking  a little closer at the doc it looks like they dropped support for multi-key sorts. so the above would need to be something like `_.sortBy(_.sortBy(results, "w"), "v")`.

Will look through the code for this pattern as I get time. Happy to take a pull request if someone wants this sooner.

**Comment by teodragovic** [comment]
Maybe try using [sortByAll](https://lodash.com/docs#sortByAll)

from v 3.0.0 changelog:
> Moved support for sorting by multiple properties from _.sortBy to _.sortByAll

**Comment by cpettitt** [comment]
Perfect! That should be straightforward to fix tonight. Thanks teodragovic!

**Comment by cpettitt** [comment]
Fixed in v0.7.3
